bson-adt-core/src/main/scala/adt/bson/mongo/codecs/BsonAdtCodecRegistry.scala

```diff
@@ -0,0 +1,19 @@
+package adt.bson.mongo.codecs
+
+import adt.bson.{BsonValue, BsonObject}
+import org.bson.codecs.Codec
+import org.bson.codecs.configuration.{CodecConfigurationException, CodecRegistry}
+
+object BsonAdtCodecRegistry extends CodecRegistry {
```

Comment:
Please document when and why you should use this registry.

Also, might want to explain why this doesn't need clean up, why this isn't a memory leak or something.